### PR TITLE
TextInput: Updated v.1 

### PR DESCRIPTION
**Pull request by raynesmike**
Following added:
added Icons to the text input
separated a working utility for phone type
useRef to focus textinput when clicking anywhere in the component
password security sensitivity pressIn/pressout
TextinputV1

### CODE REVIEW

**Review by carizzuti — APPROVED** [review]

File: mobile/src/components/textInput/TextInput.tsx [hunk 1]
```diff
@@ -19,55 +21,76 @@ export const TextInput: FC<TTextInputProps> = ({
 }) => {
   const [isActive, setIsActive] = useState(Boolean);
```
Comment (lasharela) [inline mobile/src/components/textInput/TextInput.tsx:22]: here you are setting by default value. if you want to set by default type you have to do 

useState<boolean>(false) (Do we need capital letter for types? )

File: mobile/src/components/textInput/TextInput.tsx [hunk 2]
```diff
@@ -19,55 +21,76 @@ export const TextInput: FC<TTextInputProps> = ({
...
+      currentContainerStyle = {
+        ...currentContainerStyle,
+        justifyContent: 'flex-start',
+      };
+      break;
+    default:
+      break;
+  }
+
+  useLayoutEffect(() => {
```
Comment (lasharela) [inline mobile/src/components/textInput/TextInput.tsx:66]: I would say you have to create 2 components. `TextInput` and `TextInputPassword`...

File: mobile/src/utilities/phone_validator.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
+export const phone_validator = (input: string) => {
```
Comment (lasharela) [inline mobile/src/utilities/phone_validator.ts:1]: what is return type of this method?

**Review by lasharela — CHANGES_REQUESTED** [review]

File: mobile/src/utilities/phone_validator.ts (hunk 3 above)
Comment (raynesmike) [reply mobile/src/utilities/phone_validator.ts:1]: Thank you, Ill makesure to add that

File: mobile/src/components/textInput/TextInput.tsx (hunk 1 above)
Comment (raynesmike) [reply mobile/src/components/textInput/TextInput.tsx:22]: That make sense, for readability. Thank you

File: mobile/src/components/textInput/TextInput.tsx (hunk 2 above)
Comment (raynesmike) [reply mobile/src/components/textInput/TextInput.tsx:66]: Is this a bad practice to create 1 component for the textInput? I just added a logic to set isSensitive(securedText) to hide the password

**Comment by Dulaya** [comment]
- The biggest I saw was: Do you have a way to retrieve the input value? I know you can use ...rest. However, whoever will build the screens will get type errors especially if they pass down useState function. One way to solve this problem is to explicitly create a way to pass down the function in addition to using rest.
- Another issue is that the phone number input is very buggy. 
- Some random type errors (Maybe it's just on my local development).
- Lastly, the text input and icons are really big on my screen. See picture.
- Overall, everything else work fine. I had no issues with the Email, Password and Search input. Good work.

image

**Review by Nap1400 — APPROVED** [review]

**Review by aahenry51 — APPROVED** [review]

**Pull request merged**